.github/ISSUE_TEMPLATE/enhancement-new-rule.yaml

```diff
@@ -0,0 +1,69 @@
+name: "Enhancement: Propose a New Rule"
+description: "Propose a new lint rule"
+title: "Rule proposal: a short description of my proposal"
+labels:
+  - "enhancement: new plugin rule"
+  - "package: eslint-plugin"
+  - triage
+body:
+  - type: checkboxes
+    id: sanity-checks
+    attributes:
+      label: Before You File a Proposal Please Confirm You Have Done The Following...
+      options:
+        - label: I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+plugin+rule%22) and found none that match my proposal.
+          required: true
+        - label: I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
+          required: true
+        - label: I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.
+          required: true
+  - type: checkboxes
+    id: rule-fits-the-brief
+    attributes:
+      label: My proposal is suitable for this project
+      options:
+        - label: My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
+          required: true
+        - label: My proposal is not a "formatting rule"; meaning it does not just enforce how code is formatted (white space, brace placement, etc).
```

Comment:
> white space

Nit: I've always seen us & others write it as "whitespace", no space. Intentional?